pkg/envd/docker.go

```diff
@@ -459,6 +463,23 @@ func (e dockerEngine) StartEnvd(ctx context.Context, so StartOptions) (*StartRes
 	if so.ShmSize > 0 {
 		hostConfig.ShmSize = int64(so.ShmSize) * 1024 * 1024
 	}
+	// resource
+	if len(so.NumCPU) > 0 {
+		cpu, err := strconv.ParseFloat(so.NumCPU, 64)
+		if err != nil {
+			logger.Infof("parse `cpu` error: %v, ignore this argument", err)
+		} else {
+			hostConfig.NanoCPUs = int64(cpu * 10e9)
```

Comment:
I prefer not to use the `NanoCPUs`. I think  I would be better to set the  CPU quota by using `CPUPeriod` and `CPUQuota`. I think it would be better to get less CPU throttle.

Reply:
The main reason I choose `NanoCPUs` is because it's applicable to all platforms. But I guess UNIX is also okay. WDYT cc @gaocegege

Reply:
Or we can detect the platform and apply `CPUPeriod` for UNIX.

Reply:
> Or we can detect the platform and apply `CPUPeriod` for UNIX.

SGTM